docs/config-import.md

```diff
@@ -56,6 +56,20 @@ To import the Magento configuration settings for ([@therouv](https://github.com/
 php bin/magento config:data:import config/store dev/therouv
 ```
 
+### Environment Variables substitution
+
+If you do not want to store your secrets in version control, you can use placeholders for environment variables in the configuration files. This is done with the notation `{env:ENV_VAR_NAME}`.
+
+For example, this might be the content of your config file:
+
+```
+vendorx/general/api_key:
+  default:
+    0: {env:VENDORX_API_KEY}
```

Comment:
Regarding the format, again just FYI: In https://www.mkdocs.org/user-guide/configuration/#environment-variables it presents a way to mark this with Yaml-Tags. The Yaml parser might support this already (and they also have a default fallback value which I find sweet normally). Not saying it should be done the same (that project is in python), but you may already get some aspects / suggestions. And from top of my head, I thing the symfony yaml parser is in use and they may have environment variable support already. So perhaps also worth to look into.

- https://symfony.com/doc/current/components/yaml.html#parsing-and-dumping-custom-tags
- https://symfony.com/doc/current/configuration/env_var_processors.html

Reply:
Yeah Symfony makes use of `%env()` instead of the proposed `{env()}`. Keeping in line with the original Symfony syntax makes more sense to me too.

Reply:
And I guess they have a component for that already. But take care with the deps ;) The tagged variant looks like it could work with the symfony YAML component standalone. and YAML tags have the benefit that other yaml parsers can deal with it as well. so for portability I'd personally would favor that. Then you don't need any extra syntax.

Reply:
I guess you mean [Symfony\Component\DependencyInjection\EnvVarProcessor](https://github.com/symfony/dependency-injection/blob/ffaa7b47a728b8a644ced7dce6e3f7b89b1c06bb/EnvVarProcessor.php) but I don't see a way to have that do what we're looking for.

Reply:
But I agree it would be _nice_ to have complete parity with how Symfony does it, but it is not strictly needed to have the basic functionality to support simple string env vars merged into this extension.

Reply:
Yeah nice but then hogs/problems with deps is not nice. As Symfony is popular, I'm in good hope the format is already well understood which I think is a benefit. I think this PR starts to take off.

Reply:
Disallow list;

- `PHP_`
- `HTTP_`
- `SERVER_`
- `SCRIPT_`
- `QUERY_`
- `DOCUMENT_`